optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/planner/RouteOptimizerConfig.java

```diff
@@ -36,15 +41,14 @@
         this.solverFactory = solverFactory;
     }
 
-    @Bean
+    @Produces
     Solver<VehicleRoutingSolution> solver() {
         return solverFactory.buildSolver();
     }
 
-    @Bean
-    AsyncListenableTaskExecutor executor() {
-        SimpleAsyncTaskExecutor executor = new SimpleAsyncTaskExecutor();
-        executor.setConcurrencyLimit(1);
-        return executor;
+    @Produces
+    ListeningExecutorService executor() {
+        ExecutorService executorService = Executors.newFixedThreadPool(1);
```

Comment:
The executor is not being shut down AFAIK. I don't shut it down myself.

I gave +1 to the question because I understand resources should be freed when not needed. What baffles me is

1. Where should this executor be shut down. I can do it in a `@PreDestroy` method of the SolverManager bean or I can observe the `ShutDown` event in the config class that produces the executor bean. That would me more appropriate if multiple beans used the executor. I'd say these two are equivalent at this point.
2. More importantly, **what is the risk of not doing that**? I can't see any because the executor should remain open for the whole life of the app so I can't see the **benefit of shutting down the executor just before the application stops**.

So, should I shut it down A) to mitigate a risk that I don't see (please educate me), B) because it's a good practice or C) do not shut it down to avoid unnecessary code?